### DISCUSSION TITLE
Being able to run `bisq-desktop` and `bisq-daemon` in one process

### DISCUSSION THREAD

**Discussion by unbeaten69** [bisq-network/bisq#7209] [Ideas]
When integrating with Bisq via the API it would be really helpful to be able to see the results of operations triggered via the API in the frontend, either immediately or almost immediately. But currently it does not seem to be possible, I have to choose between running `bisq-desktop` or `bisq-daemon`. Whenever I make some API call to `bisq-daemon` and want to see what has actually happened, the only option seems to be to shut the daemon down and start the desktop app. Then I let's say explore the order that I have just placed etc. Or cancel it manually so that I can make another attempt of placing it via the API.

This is how the APIs of basically all the centralized exchanges work, it is possible see the outcomes of API actions in their frontends in real time. That makes trial-and-error integration significantly easier, simpler, faster.

**Comment by ghubstan** [comment]
You cannot run both in the same process.  With care (configuration), you _could_ run both processes on the same host, but _don't do it_.   The `bisq-desktop` and `bisq-daemon` must each have their own wallets.  You can either run `bisq-desktop` and `bisq-daemon`  on separate hosts, or configure `bisq-daemon` to point to a different wallet (on the same host as  `bisq-desktop`).  

Running both on the same host, pointing to the same wallet's will lead to wallet corruption.

**Reply by unbeaten69** [reply]
Yeah, this is all clear. What I meant is that it would be nice if in the future the main process (`bisq-desktop` at the moment, or whatever takes over its role in the future) could expose the trading API at the same time, in such a way that actions triggered via that API get reflected in the frontend immediately. This would noticeably improve ease of integration.
